runtime/makelib/targets.mk.ftl

```diff
@@ -238,14 +236,23 @@ ifdef UMA_ADD_COPYRIGHT
   UMA_OBJECTS:=$(UMA_COPYRIGHT_FILE)$(UMA_DOT_O) $(UMA_OBJECTS)
 endif
 
-<#if uma.spec.type.unix>
+<#if uma.spec.type.unix && !uma.spec.type.ztpf>
 UMA_LINK_PATH+=-L. -L$(UMA_PATH_TO_ROOT) -L$(UMA_PATH_TO_ROOT)lib/
 UMA_BUILD_LIBRARIES:=$(foreach lib,$(UMA_LIBRARIES),$(if $($(lib)_alllib),$(lib)))
 UMA_LINK_PATH+=$(foreach lib,$(UMA_BUILD_LIBRARIES),-L$($(lib)_path))
 # Remove duplicate paths
 UMA_LINK_PATH:=$(sort $(UMA_LINK_PATH))
 </#if>
 
+<#if uma.spec.type.ztpf>
+UMA_LINK_PATH+=-L. -L$(UMA_PATH_TO_ROOT) -L$(UMA_PATH_TO_ROOT)lib/
+UMA_BUILD_LIBRARIES:=$(foreach lib,$(UMA_LIBRARIES),$(if $($(lib)_alllib),$(lib)))
+UMA_LINK_PATH+=$(foreach lib,$(UMA_BUILD_LIBRARIES),-L$($(lib)_path))
+# removed sorting of UMA_LINK_PATH which is done for all unix platforms (see similar unix block above)
+# z/TPF unique build environment is sensitive to link order for some cases.
+#UMA_LINK_PATH:=$(sort $(UMA_LINK_PATH))
+</#if>
```

Comment:
Rather than duplicating this whole block, can you #if just the sorting?

```
<#if uma.spec.type.unix>
UMA_LINK_PATH+=-L. -L$(UMA_PATH_TO_ROOT) -L$(UMA_PATH_TO_ROOT)lib/
UMA_BUILD_LIBRARIES:=$(foreach lib,$(UMA_LIBRARIES),$(if $($(lib)_alllib),$(lib)))
UMA_LINK_PATH+=$(foreach lib,$(UMA_BUILD_LIBRARIES),-L$($(lib)_path))
<#if !uma.spec.type.ztpf>
# Remove duplicate paths 
UMA_LINK_PATH:=$(sort $(UMA_LINK_PATH))
<#endif>
```